harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -143,18 +143,51 @@ def _set_data_loaders(self) -> None:
     def run(self) -> None:
         for w, args, response_func in self.workloads:
             if w.kind == workload.Workload.Kind.RUN_STEP:
-                response_func(
-                    util.wrap_metrics(
-                        self._train_for_step(w.step_id, w.num_batches, w.total_batches_processed),
-                        self.context.get_stop_requested(),
+                try:
+                    response_func(
+                        util.wrap_metrics(
+                            self._train_for_step(
+                                w.step_id, w.num_batches, w.total_batches_processed
+                            ),
+                            self.context.get_stop_requested(),
+                            self.context.get_invalid_hp(),
+                        )
                     )
-                )
+                except det.InvalidHP as e:
+                    logging.info(
+                        "Invalid hyperparameter exception in trial train step: {}".format(e)
+                    )
+                    response_func(
+                        util.wrap_metrics(
+                            self._train_for_step(
+                                w.step_id, w.num_batches, w.total_batches_processed
+                            ),
+                            self.context.get_stop_requested(),
+                            self.context.set_invalid_hp(True),
+                        )
+                    )
+                    raise
             elif w.kind == workload.Workload.Kind.COMPUTE_VALIDATION_METRICS:
-                response_func(
-                    util.wrap_metrics(
-                        self._compute_validation_metrics(), self.context.get_stop_requested()
+                try:
```

Comment:
I think you actually want to put only `self._compute_validation_metrics()` in the "try" rather than the call to `response_func`.

Reply:
I think since the `except` clause only catches InvalidHP, and since InvalidHP is only thrown by user code, and since the response_func call is quite different in the two cases, I think this is actually a pretty reasonable strategy.

Reply:
> since the response_func call is quite different in the two cases, I think this is actually a pretty reasonable strategy.

I don't understand this. We only run user code in `self._compute_validation_metrics()`, which should be the only place to throw InvalidHP. How would response_func interact with InvalidHP?

Reply:
If you look at `_workload_manager.py`, I'm handling InvalidHP as a special case when handling the response_func, and that is how information is being propagated to the searcher to handle InvalidHP